src/main/java/com/library/java_library_management/repository/GeneralModeRepository.java

```diff
@@ -0,0 +1,181 @@
+                fileControl.modifyFile(file.getAbsolutePath(), "status", jsonStatus);
+                scheduler.schedule(() -> {
+                    try {
+                        String statusAfterFive = objectMapper.writeValueAsString(BookStatus.AVAILABLE);
+                        fileControl.modifyFile(file.getAbsolutePath(), "status", statusAfterFive);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }, 5, TimeUnit.MINUTES);
+                System.out.println("반납 처리 완료 되었습니다.");
+            }
+        }catch (IOException e){
+            e.printStackTrace();
+        }catch (RuntimeException e){
+            System.out.println("원래 대여 가능한 도서입니다");
+        }
+    }
+
+
+    @Override
+    public void missBook(int book_id) {
+        try{
+            File file = new File(fileInfo.getFilePath() + book_id + ". book.json");
+            BookInfo book = getBookFromFile(file);
+            if(book.getStatus() == BookStatus.LOST)
+                throw new RuntimeException();
+            else{
+                String jsonStatus = objectMapper.writeValueAsString(BookStatus.LOST);
+                fileControl.modifyFile(file.getAbsolutePath(), "status", jsonStatus);
+                System.out.println("분실 처리 완료 되었습니다.");
+            }
+        }catch (IOException e){
+            System.out.println("존재하지 않는 도서입니다.");
+        }
+        catch (RuntimeException e){
+            System.out.println("이미 분실 처리된 도서입니다.");
+        }
+    }
+
+    @Override
+    public void deleteById(int book_id) {
+
+        try{
+            File file = new File(fileInfo.getFilePath() + book_id + ". book.json");
+            Files.delete(Paths.get(file.getAbsolutePath()));
+            System.out.println("삭제처리 되었습니다.");
+        }catch (IOException e){
+            System.out.println("존재하지 않는 도서입니다.");
+        }
+    }
+
+
+    private BookInfo getBookFromFile(File file) throws IOException {
+        String status = fileControl.readFile(file.getAbsolutePath(), "status");
+        BookStatus bookStatus = objectMapper.readValue(status, BookStatus.class);
+
+        int book_id = Integer.parseInt(fileControl.readFile(file.getAbsolutePath(), "book_id"));
+        String title = fileControl.readFile(file.getAbsolutePath(), "title");
+        String author = fileControl.readFile(file.getAbsolutePath(), "author");
+        int page_size = Integer.parseInt(fileControl.readFile(file.getAbsolutePath(), "page_size"));
+
+        BookInfo book = new BookInfo(book_id, title, author, page_size, bookStatus);
+        return book;
+    }
+
+    //2번 모드 구현 때, 테스트 코드 작성용 메소드
+    @Override
+    public Optional<BookInfo> findSameBook(String title) {
+        List<BookInfo> totalBook = getTotalBook();
+        for(BookInfo book : totalBook){
+            if(book.getTitle().equals(title)){
+                return Optional.ofNullable(book);
```

Comment:
```suggestion
                return Optional.of(book);
```

Optional.of 와 ofNullable의 의미가 다른데 한번 알아볼까요?

Reply:
null 값 허용유무 차이로 알고 있는데 예제 코드 보며 더 확실히 이해해두겠습니다!